### PR TITLE
s3: small cleanup and limit memory usage by disabling threads on upload_fileobj

### PR DESCRIPTION
**Pull request by pazembrz**

### CODE REVIEW

File: backend/inspirehep/records/api/literature.py [hunk 1]
```diff
@@ -305,6 +305,7 @@ def add_file(
         new_key = hash_data(file_data)
         mimetype = magic.from_buffer(file_data, mime=True)
         size = len(file_data)
+        file_data = BytesIO(file_data)
```
Comment (michamos) [inline backend/inspirehep/records/api/literature.py]: what's the point of this?

Comment (pazembrz) [reply backend/inspirehep/records/api/literature.py]: Here yes. Main part is disabling threads on `upload_fileobj`.

**Comment by drjova** [comment]
do we have any metrics for this change? How much memory it's saving?

**Comment by drjova** [comment]
TBH I'm not convinced about this change, let's discuss it IRL

**Pull request merged**